packages/block-library/src/button/editor.scss

```diff
@@ -40,7 +40,6 @@
 		.wp-block-button__link {
 			max-width: 100%;
 			overflow: hidden;
-			white-space: nowrap;
```

Comment:
I don't think this one should be removed. It was written to be applied _only_ when the button is shown as a preview, for example in the variation interface. Its purpose, along with `text-overflow: ellipsis;` is to ensure that the preview looks good. 

Does this rule ever bleed outside of `.block-editor-block-preview__content`?

I understand that the `!important` rule will override it anyway, so that would break it too.

If this wrapping rule bleeds out of `.block-editor-block-preview__content` then that seems to be the rule to fix, otherwise we regress what #10565 tried to fix.